lib/agriculture_sensor_parser.ex

```diff
@@ -0,0 +1,354 @@
+              x: x_achis,
+              y: y_achis,
+              z: z_achis
+            }
+          ),
+          overhead # pass the overhead..
+        )
+        
+        
+      # impact_magnitude | 2 Byte | Fakt: 1 milli-g
+      << 0x0A, 0x02, impact_magnitude::16 , overhead::binary >> ->  
+        parse_dyn(
+          Map.put_new(
+            parsed_payload,               
+            :impact_magnitude,
+            impact_magnitude
+          ),
+          overhead # pass the overhead..
+        )
+        
+        
+      # 1. impact_alarm [FALSE] | 1 Byte | Fakt: Boolean (0x00 -> No impact alarm; 0xFF -> ALARM!)
+      << 0x0A, 0x00, 0x00, overhead::binary >> ->  
+        parse_dyn(
+          Map.put_new(
+            parsed_payload,               
+            :impact_alarm,
+            false
+          ),
+          overhead # pass the overhead..
+        )
+      # 2. impact_alarm [TRUE]
+      << 0x0A, 0x00, 0xFF, overhead::binary >> ->  
+        parse_dyn(
+          Map.put_new(
+            parsed_payload,               
+            :impact_alarm,
+            true
+          ),
+          overhead # pass the overhead..
+        )
+        
+        
+      # ambient_temperature | 2 Byte | Fakt: 0.1 °C/LSB signed
+      << 0x0B, 0x67, ambient_temperature::integer-size(16), overhead::binary >> ->  
+        parse_dyn(
+          Map.put_new(
+            parsed_payload,               
+            :ambient_temperature,
+            ambient_temperature * 0.1  # scaled
+          ),
+          overhead # pass the overhead..
+        )
+        
+        
+      # ambient_relative_humidity | 1 Byte | Fakt: 0.5%/LSB
+      << 0x0B, 0x68, ambient_relative_humidity::8 , overhead::binary >> ->  
+        parse_dyn(
+          Map.put_new(
+            parsed_payload,               
+            :ambient_relative_humidity,
+            ambient_relative_humidity * 0.5  # scaled
+          ),
+          overhead # pass the overhead..
+        )
+        
+        
+      # mcu_temperature | 1 Byte | Fakt: 0.1 °C/LSB signed
+      << 0x0C, 0x67, mcu_temperature::integer-size(8), overhead::binary >> ->  
+        parse_dyn(
+          Map.put_new(
+            parsed_payload,               
+            :mcu_temperature,
+            mcu_temperature * 0.1  # scaled
+          ),
+          overhead # pass the overhead..
+        )
+        
+        
+      # Failsafe / Stop point (if no condition fits or data is just empty..)
+      << x::binary >> when byte_size(x) > 0 ->
+        Logger.error "There is unparsed data left! (#{byte_size(x)} Bytes) - It could be invalid or an parsing error.."
+        parsed_payload = Map.put_new(parsed_payload, :configuration_control_commands, "There is unparsed data left! (#{byte_size(x)} Bytes) => (#{Base.encode16(x)})")  # To see in Control/Field
+        
+        [parsed_payload]
+        
+      # Default behaviour -> no bytes left
+      _ -> 
```

Comment:
Can only be `<<>>`, the empty binary.

Reply:
The underscore is the placeholder for any value, equal to "default" in other languages, isn't it? 
Or did you ment the relevance for the upper part `<< x::binary >> when byte_size(x) > 0 ->`?
It's an possibility, that in future firmware releases of the agriculture sensor the Payload might extend,
for this case I print an error with the rest of the Payload itself as a part of the returned error. 
I thought it would be an greater solution than just silent fail or just log it (when now one is watching logs).

If you ment the default behaviour, the empty Bitstring `<<>>` is also possible, but in the style of the case element, I would preffer the Underscore `_`, for this case I'll switch it to the Bitstring. 💪